benchadapt/python/benchadapt/adapters/_adapter.py

```diff
@@ -17,16 +17,42 @@ class BenchmarkAdapter(abc.ABC):
     command : List[str]
         A list of args to be run on the command line, as would be passed
         to `subprocess.run()`.
+    result_defaults_override : Dict[str, Any]
+        A dict of default values to be passed to `BenchmarkResult`. Useful for
+        specifying metadata only available at runtime, e.g. build info. Overrides
+        `BenchmarkResult` defaults; is overridden by values passed in
+        ``transform_results()``. Values of ``None`` do not unset defaults, just as
+        when passed to ``BenchmarkResult`'s init method.
+    results_defaults_append : Dict[str, Any]
+        A dict of default values to be appended to `BenchmarkResult` values.
+        Appended after instantiation. Useful for appending extra tags or other
+        metadata in addition to that gathered elsewhere. Only applicable for dict
+        attributes. For each element, will override any keys that already exist,
+        i.e. it does not append recursively.
     results : List[BenchmarkResult]
         Once `run()` has been called, results from that run
     """
 
     command: List[str]
+    result_defaults_override: Dict[str, Any] = None
```

Comment:
`result_defaults_append` should go here too right?

Reply:
yep, realized this later...not quite sure how this was working?

Reply:
Technically you don't need to declare any of these attributes at all, at least if you set them before you try to read them. But it's nice for type hinting and so devs can keep track of what attributes you're planning to use in methods.